src/App.jsx

```diff
@@ -14,33 +16,78 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getSorted(obj, way) {
+  const currentGoods = [...obj];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  currentGoods.sort((el1, el2) => {
+    switch (way) {
+      case 'is-info':
+        return el1.localeCompare(el2);
+      case 'is-success':
+        return el1.length - el2.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return currentGoods;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [selectItem, setSelectItem] = useState('');
+  const [selReverse, setselReverse] = useState('');
+  const goods = getSorted(goodsFromServer, selectItem);
+
+  function reset() {
+    setselReverse('');
+    setSelectItem('');
+  }
+
+  if (selReverse) {
```

Comment:
This is a performance issue. You are reversing the `goods` array directly in the component body. This operation will be performed on every render, even when `selReverse` hasn't changed. It would be better to do this operation in a useMemo hook, so it only happens when `goods` or `selReverse` changes.